guppylang/prelude/builtins.py

```diff
@@ -55,6 +55,16 @@ def py(*_args: Any) -> Any:
     raise GuppyError("`py` can only by used in a Guppy context")
 
 
+class _Inout:
+    """Dummy class to support `@inout` annotations."""
```

Comment:
This is to allow `def foo(x: int @inout)`, since Python (in its infinite wisdom *cough* *splutter* *boo*) disallows `def foo(x: @inout int)`. Obviously, it's fine to do *matrix multiplication* inside a type annotation but not invoke a decorator....

However, have you thought about the "pythonic" alternative, which I believe would be `def foo(x: Annotated[int, "inout"]`?`@inout` is a bastardization of python syntax anyway so what would users think? I admit `Annotated` is rather long - what about `def foo(x: Inout[int])` ? (No way, I hear you say - and I see the downside - maybe that lengthy Annotated isn't sooo bad, then....)

Not seriously recommending you change, but how different / how much simpler might that make things, would it impact the design at all?

Reply:
I created  #359 for discussion.

> Not seriously recommending you change, but how different / how much simpler might that make things, would it impact the design at all?

I think this is mostly a syntax concern. The design wouldn't be affected much imo